examples/app/bokeh-decision-tree/Decision_Tree/ID3_Decision_Tree/bucheim.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+#####################################################################################################
+# Implementation of "Drawing rooted trees in linear time(Buchheim, Junger and Leipert, 2006)"#
+
+# constant for distance between two nodes
+distance = 5
+
+
+def tree_layout(node):
+    """ main function """
+    first_walk(node)
+    second_walk(node, 0-node.mod)
+
+
+def first_walk(node):
+    """
+        Calling FIRSTWALK(node) computes a preliminary x-coordinate for node. Before that, FIRSTWALK is
+        applied recursively to the children of node, as well as the function APPORTION. After spacing out the
+        children by calling EXECUTESHIFTS, the node  is placed to the midpoint of its outermost children.

```

Comment:
Docstrings in Bokeh typically use triple single quotes, i.e. `'''` and docstring always starts on the same line as the first triple quotes. Ideally that line is a single short intro/decarative sentence, and then after a blank line, any additional details. We use google style parameter docs. Here is a prototypical example including:

```
def find(objs, selector, context=None):
    ''' Query a collection of Bokeh models and yield any that match a selector.

    Queries are specified as selectors similar to MongoDB style query
    selectors, as described for :func:`~bokeh.core.query.match`.

    Args:
        obj (Model) : object to test
        selector (JSON-like) : query selector
        context (dict) : kwargs to supply callable query attributes

    Returns:
        Model : objects that match the query
    
    '''
```

Reply:
We will fix these as soon as possible.